modules/tinymce/src/core/main/ts/selection/DetailsElement.ts

```diff
@@ -105,6 +119,54 @@ const preventDeletingSummary = (editor: Editor): void => {
         e.preventDefault();
         removeNode(editor, node);
         CaretFinder.firstPositionIn(nextNode).each(setCaretToPosition(editor));
+      } else if (isSafari && isSummary(node)) {
+        // TINY-9951: Safari has a bug where upon pressing Backspace/Delete when the caret is directly within the summary,
+        // all content is removed and the caret is prevented from being placed back into the summary.
+        e.preventDefault();
+
+        if (!isCollapsed && isEntireNodeSelected(rng, node) || DeleteUtils.willDeleteLastPositionInElement(isDelete, CaretPosition.fromRangeStart(rng), node)) {
+          emptyNodeContents(node);
+        } else {
+          // Wrap all summary children in a temporary container to execute Backspace/Delete there, then unwrap
+          const sel = selection.getSel();
+          let { anchorNode, anchorOffset, focusNode, focusOffset } = sel ?? {};
+          const applySelection = () => {
+            if (Type.isNonNullable(anchorNode) && Type.isNonNullable(anchorOffset) && Type.isNonNullable(focusNode) && Type.isNonNullable(focusOffset)) {
+              sel?.setBaseAndExtent(anchorNode, anchorOffset, focusNode, focusOffset);
+            }
+          };
+          const updateSelection = () => {
+            anchorNode = sel?.anchorNode;
+            anchorOffset = sel?.anchorOffset;
+            focusNode = sel?.focusNode;
+            focusOffset = sel?.focusOffset;
+          };
+          const appendAllChildNodes = (from: Node, to: Node) => {
+            Arr.each(from.childNodes, (child) => {
+              if (FormatUtils.isNode(child)) {
+                to.appendChild(child);
+              }
+            });
+          };
+
+          const container = editor.dom.create('span');
+          appendAllChildNodes(node, container);
+          node.appendChild(container);
+          applySelection();
+          // Manually perform ranged deletion by keyboard shortcuts
+          if (isCollapsed && (isMacOS && (e.altKey || isBackspace && e.metaKey) || !isMacOS && e.ctrlKey)) {
```

Comment:
We already know the browser is MacOS right from the parent `if` statement assuming I reading it correctly.

Reply:
My thought process was that the `isSafari` in the parent if statement doesn't explicitly cover `isMacOS`. That is if one day they decide to bring Safari to other platforms.

Reply:
This also makes me wonder if I should check if it is `isiOS` as well, since these shortcuts can also work on those devices with a keyboard connected.

Reply:
Enabled the overrides for iOS.